### ISSUE TITLE
bad behaviour on draft deletion: incorrect set of discussion_id

### ISSUE THREAD

**Issue by iamdey** [CaliOpen/Caliopen#1237]
create a reply to a discussion:

`POST http://localhost:4000/api/v1/messages`

```
{
	"message_id": "dcc1786c-bc61-4ca5-8517-96d86c1629bd",
	"subject": "Johan sent you a new message",
	"body": "avec discussion_id\n",
	"discussion_id": "581eefc9-226a-47d9-a4a7-f50937881492",
	"parent_id": "607dc5cd-d948-4391-a786-825c5542011a",
	"user_identities": ["21ac7f0f-e88f-47b9-beb3-8c267e23a7d2"]
}
```

But afterward the discussion_id is not set

`http://localhost:4000/api/v2/messages/dcc1786c-bc61-4ca5-8517-96d86c1629bd`

```
{
	"body": "avec discussion_id\n",
	"body_is_plain": true,
	"excerpt": "avec discussion_id\n",
	"date": "2019-03-22T09:41:56.118Z",
	"date_insert": "2019-03-22T09:41:56.118Z",
	"date_sort": "2019-03-22T09:41:56.118Z",
	"user_identities": ["21ac7f0f-e88f-47b9-beb3-8c267e23a7d2"],
	"importance_level": 0,
	"is_answered": false,
	"is_draft": true,
	"is_unread": false,
	"is_received": false,
	"message_id": "dcc1786c-bc61-4ca5-8517-96d86c1629bd",
	"parent_id": "607dc5cd-d948-4391-a786-825c5542011a",
	// …
	"subject": "Johan sent you a new message",
	"protocol": "email",
	"user_id": "42cd4848-edf0-458b-80e2-42ca48437658"
}
```
---
The client uses the discussion_id afterward so, 2 possibilities:
 
* The `discussion_id` should be untouched in case it is defined by the client: https://trello.com/c/pRPw9abA/165-edit-participants-calc-discussion
* The `discussion_id` is a read only prop, it can be calculated by the client because a `parent_id` is mandatory for a reply

**Comment by iamdey** [comment]
after discussion with @gdchamal, we will use `parent_id` to calc the `discussion_id`

It will result in multiple fetches in draft view

**Comment by iamdey** [comment]
there is still an issue when the user try to retrieve a reply:

on a discussion, the client try to initialize the draft by fetching `/messages?discussion_id=xxx&is_draft=true&limit=1` but nothing is return.

Solution to get rid of this: list all drafts that has a parent_id, fetch the parent to get the associated discussion_id.

I'm not super fan of this.

**Comment by iamdey** [comment]
should be fixed in 0.18 for current data model
and will be verified in #1278
